Formula/unisonlang.rb

```diff
@@ -0,0 +1,76 @@
+require "language/node"
+
+class Unisonlang < Formula
+  desc "Friendly programming language from the future"
+  homepage "https://unison-lang.org/"
+  url "https://github.com/unisonweb/unison.git",
+      tag:      "release/M3",
+      revision: "97cce8d23147df30b0f80ae745968db09f4e1a44"
+  version "M3"
+  license "MIT"
+  head "https://github.com/unisonweb/unison.git", branch: "trunk"
+
+  depends_on "ghc" => :build
+  depends_on "haskell-stack" => :build
+  depends_on "node" => :build
+
+  uses_from_macos "ncurses"
+  uses_from_macos "zlib"
+
+  on_linux do
+    depends_on "xz" => :build
+    depends_on "gmp"
+  end
+
+  resource "unison-ui" do
+    url "https://github.com/unisonweb/codebase-ui.git",
+        tag:      "release/M3",
+        revision: "05df3a235dec7d4a03ff1c588a32eb6daca288d4"
+  end
+
+  def install
+    jobs = ENV.make_jobs
+    ENV.deparallelize
+
+    ghc_args = [
+      "--system-ghc",
+      "--no-install-ghc",
+      "--skip-ghc-check",
+      "--copy-bins",
+      "--local-bin-path=#{buildpath}",
+    ]
+
+    system "stack", "-j#{jobs}", "build", "--flag", "unison-parser-typechecker:optimized", *ghc_args
+
+    cp buildpath/"unison", prefix/"ucm"
```

Comment:
Any reason why we're not doing this
```suggestion
    prefix.install "unison" => "ucm"
```
?

It is because we need to keep `unison` in `buildpath` for the succeeding parts of the build? If so, why not just delay the `prefix.install` to the very end?

Reply:
I just forgot `.install` can take a map 😝